webapp/stores/user_store.jsx

```diff
@@ -105,6 +105,10 @@ class UserStoreClass extends EventEmitter {
         this.currentUserId = user.id;
         global.window.mm_current_user_id = this.currentUserId;
         if (LocalizationStore.getLocale() !== user.locale) {
+            if (!GlobalActions) {
```

Comment:
Just deleted a paragraph explaining why I didn't think we should do that now because each Post component would need the list of all profiles for highlighting at mentions and checking that in shouldComponentUpdate would be really slow BUT I realized the way we do it right now doesn't do anything with shouldComponentUpdate anyway.

I'm going to take another shot at getting rid of UserStore in TextFormatting, but I'm not going to bother checking the profile list in PostMessageView.shouldComponentUpdate. It won't be perfect since the profile list changing won't trigger the PostMessageView to rerender, but that's no different than it works currently, and it'll fix the dependency issue.